Lib/test/_test_multiprocessing.py

```diff
@@ -6155,3 +6155,9 @@ class SemLock(_multiprocessing.SemLock):
         name = f'test_semlock_subclass-{os.getpid()}'
         s = SemLock(1, 0, 10, name, False)
         _multiprocessing.sem_unlink(name)
+
+    def test_semlock_mixed_context(self):
+        queue = multiprocessing.get_context("fork").Queue()
+        p = multiprocessing.get_context("spawn").Process(target=close_queue, args=(queue,))
```

Comment:
Do we also want to test with "forkserver" method?

Reply:
Good catch! Added it.
It now ensure that all cases work. Let me know if that properly covers all the cases you had in mind.